python/importer/main.py

```diff
@@ -8,32 +8,63 @@
 from datetime import datetime, timedelta
 import psycopg2 as psycopg
 from common.utils import get_conn_params, get_logger
+import common.constants as constants
 from .run_analysis import main as run_analysis
 
 # TODO: import other event types as well when needed.
 event_types_to_import = ['DOC', 'DOO']
 
 def main(dataImporter: func.TimerRequest):
     logger = get_logger()
-    logger.info("### Going to run importer. ###")
+
     conn = psycopg.connect(**get_conn_params())
+    global is_importer_locked
     try:
         with conn:
             with conn.cursor() as pg_cursor:
-                import_data_to_db(pg_cursor=pg_cursor)
+                # Check if importer is locked or not. We use lock strategy to prevent executing importer
+                # and analysis more than once at a time
+                pg_cursor.execute(f"SELECT is_lock_enabled({int(constants.IMPORTER_LOCK_ID)})")
```

Comment:
f-string manipulation imposes a risk of SQL injection; although there is no user input and the risk is also minimized here by forcing the input into `int()`, I would keep up the habit of not using direct string manipulation for SQL statements if only possible.

Here, we could use psycopg2 [parameter](https://www.psycopg.org/docs/usage.html#passing-parameters-to-sql-queries) tuples safely & easily: `pg_cursor.execute("SELECT is_lock_enabled(%s)", (constants.IMPORTER_LOCK_ID, ))`. I recommend updating other "f-stringed" SQL statements accordingly.

Once you need to parameterize not only values but also table or column names, it might appear a bit hairy, but there is also a [nice & safe psycopg2 method](https://stackoverflow.com/questions/13793399/passing-table-name-as-a-parameter-in-psycopg2) for that.